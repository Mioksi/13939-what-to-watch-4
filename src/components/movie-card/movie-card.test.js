import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';

import history from '../../history';

import MovieCard from './movie-card.jsx';

const film = {
  id: 1,
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  [`preview_image`]: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  [`preview_video_link`]: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

it(`Should MovieCard render correctly`, () => {
  const tree = renderer
    .create(
        <Router
          history={history}
        >
          <MovieCard
            movie={film}
            isPlaying={false}
            onCardClick={() => {}}
            onCardMouseEnter={() => {}}
            onCardMouseLeave={() => {}}
            onStartPlaying={() => {}}
            onStopPlaying={() => {}}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
