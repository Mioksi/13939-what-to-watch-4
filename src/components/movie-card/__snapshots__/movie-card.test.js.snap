// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/films/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        className="player__video"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
